### PR TITLE
auto-attach: add new return code

### PR DESCRIPTION
**Pull request by lucasmoura**
## Why is this needed?
If we can attach the machine, but we fail to enable any service during the process, we will exit with a return code of 3

## Test Steps
Verify that the new unit test is working as expected

## Checklist

 - [x] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [ ] I have updated or added any documentation accordingly

## Does this PR require extra reviews?

 - [ ] Yes
 - [x] No

### CODE REVIEW

**Review by renanrodrigo — COMMENTED** [review]
question: for this case, wouldn't it be enough to set `exit_code = 3` to the `EntitlementsNotEnabledError`?

**Comment by orndorffgrant** [comment]
@cpaelzer also mentioned using powers of 2 for exit codes as a standard practice so that we can combine them in the future if needed

**Comment by lucasmoura** [comment]
@orndorffgrant @renanrodrigo updated

File: uaclient/cli.py [hunk 1]
```diff
@@ -1452,6 +1452,9 @@ def action_auto_attach(args, *, cfg: config.UAConfig) -> int:
     except exceptions.UrlError:
         event.info(messages.ATTACH_FAILURE.msg)
         return 1
+    except exceptions.EntitlementsNotEnabledError as e:
```
Comment (renanrodrigo) [inline uaclient/cli.py:1455]: sorry I could have explained my thought better
I think this part of the code is not needed. Defining the `exit_code` for `UserFacingError` is enough for the main error handler to exit correctly ( like happens today with the `AlreadyAttachedError`, for instance)

Comment (lucasmoura) [reply uaclient/cli.py:1455]: Fair enough, good point. I will update the code

Comment (lucasmoura) [reply uaclient/cli.py]: @renanrodrigo fixed

**Review by renanrodrigo — APPROVED** [review]
lgtm

**Comment by orndorffgrant** [comment]
Bad news - this exception is only raised if you set the `enable` or `enable_beta` list args to the full_auto_attach endpoint. It won't get raised from the automatically enabled services if they fail while getting enabled.

So I think we need to modify the auto-enable code path so that it can raise the same exception.

**Comment by orndorffgrant** [comment]
I think that would involve changing the exception raised here: https://github.com/canonical/ubuntu-pro-client/blob/main/uaclient/contract.py#L467-L470

**Review by renanrodrigo — APPROVED** [review]
lgtm again, thanks for all effort

**Review by orndorffgrant — APPROVED** [review]

**Pull request merged**